nats.go

```diff
@@ -3001,11 +3001,49 @@ func (nc *Conn) Publish(subj string, data []byte) error {
 	return nc.publish(subj, _EMPTY_, nil, data)
 }
 
+// Header represents the optional Header for a NATS message,
+// based on the implementation of http.Header.
+type Header map[string][]string
+
+// Add adds the key, value pair to the header.
+// It appends to any existing values associated with key.
+func (h Header) Add(key, value string) {
```

Comment:
I would be explicit and have a one liner for each APIs that says that the key is case-sensitive. I understand that we say that in the PR, but I want user to be able to see it when using this API.

Reply:
Good idea, ok I updated the godoc. Thanks.